### DISCUSSION TITLE
Are there any cons/pros on using autoincrement() over cuid()?

### DISCUSSION THREAD

**Discussion by johhansantana** [prisma/prisma#2515] [Q&A] [answered]
I recently updated a project from v1 to v2 and totally missed the part of the `cuid()` for `@id` which made me think that I had to change the ids to start using `Int` and `autoincrement()` to be able to make it work correctly.

Now that I saw that you could use the `cuid()` for the ids I wonder if I made a mistake on changing the types to `Int`?

**Comment by Sytten** [comment, marked as answer]
The problem with the current implementation is that cuid and uuid are stored as string inside the database because there is no support for native types, Int is stored correctly though so performances are great.

**Reply by johhansantana** [reply]
So based on this answer, the recommended way would be to use `Int` autoincrement over the `String` cuid for the references/relations?

**Reply by Sytten** [reply]
That is what we do currently until we get native types then we will switch over to UUID

**Reply by nyedidikeke** [reply]
[Native types](https://github.com/prisma/prisma/releases/tag/2.11.0) now supported (preview).

**Comment by ludralph** [comment]
Hi @johhansantana,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
